### PR TITLE
enhancement #502: refatored Subject entity

### LINKED ISSUE #502: Refactor Subject entity
Your task will be to refactor all files, related to **Subject** entity, such as controller, service, responses, DTOs,  etc.

Your steps can be as following:
1. Check service and repository class for potential issues, misspeling or incorect code style.
2. You need to run unit tests for this service and try to enhance them, also check that all necessary service methods are covered.
3. Check responses and DTOs. As we have a lot of changes some methods can return and take undocumentated information.
4. After all responses and DTOs are checked you can separate documentation of endpoints in controller as made in [this](https://github.com/fictadvisor/fictadvisor/pull/497) pr.

### PR DESCRIPTION
**Pull request by ReGeNss**
- created documentation files for each SubjectController endpoint
- refactored CreateSubjectDTO, QueryAllSubjectDTO, UpdateSubjectDTO
closes #502

### CODE REVIEW

File: utils/src/requests/UpdateSubjectDTO.ts [hunk 1]
```diff
@@ -10,7 +10,9 @@ import {
 } from '../ValidationUtil';
 
 export class UpdateSubjectDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Subject name',
+  })
```
Comment (VlO-OlV) [inline utils/src/requests/UpdateSubjectDTO.ts:15]: Add IsString decorator with validation message and also add that message to documentation

File: fictadvisor-api/src/v2/utils/documentation/subject/create.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export const SubjectDocumentationCreate: ApiDocumentationParams = {
+  isAuth: true,
+  ok: {
+    type: SubjectResponse,
+  },
+  badRequest: {
+    description: `InvalidBodyException:\n 
+                  Name is too short (min: 5)
+                  Name is too long (max: 150)
+                  Name can not be empty
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/subject/create.ts:13]:
```suggestion
                  Name cannot be empty
```

File: fictadvisor-api/src/v2/utils/documentation/subject/update.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    type: SubjectResponse,
+  },
+  badRequest: {
+    description: `InvalidEntityIdException:\n 
+                  subject with such id is not found
+                  
+                  InvalidBodyException:\n 
+                  Name is too short (min: 5)
+                  Name is too long (max: 150)
+                  Name can not be empty
```
Comment (VlO-OlV) [inline fictadvisor-api/src/v2/utils/documentation/subject/update.ts:16]:
```suggestion
                  Name cannot be empty
```

File: utils/src/requests/CreateSubjectDTO.ts [hunk 4]
```diff
@@ -10,7 +10,9 @@ import {
 } from '../ValidationUtil';
 
 export class CreateSubjectDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Subject name',
+  })
   @MinLength(5, validationOptionsMsg('Name is too short (min: 5)'))
   @MaxLength(150, validationOptionsMsg('Name is too long (max: 150)'))
   @IsNotEmpty(validationOptionsMsg('Name can not be empty'))
```
Comment (VlO-OlV) [inline utils/src/requests/CreateSubjectDTO.ts:18]:
```suggestion
  @IsNotEmpty(validationOptionsMsg('Name cannot be empty'))
```

File: utils/src/requests/UpdateSubjectDTO.ts [hunk 5]
```diff
@@ -10,7 +10,9 @@ import {
...
   @MinLength(5, validationOptionsMsg('Name is too short (min: 5)'))
   @MaxLength(150, validationOptionsMsg('Name is too long (max: 150)'))
   @IsNotEmpty(validationOptionsMsg('Name can not be empty'))
```
Comment (VlO-OlV) [inline utils/src/requests/UpdateSubjectDTO.ts:18]:
```suggestion
  @IsNotEmpty(validationOptionsMsg('Name cannot be empty'))
```

File: utils/src/requests/QueryAllSubjectDTO.ts [hunk 6]
```diff
@@ -49,4 +49,4 @@ export class QueryAllSubjectDTO {
   })
   @IsOptional()
     groupId?: string;
-}
\ No newline at end of file
+}
```
Comment (VlO-OlV) [inline utils/src/requests/QueryAllSubjectDTO.ts:52]: This dto must inherit properties from QueryAllDTO, then some of the properties here can be deleted (page, pageSize, search, order). After that you need to update the messages in documentation.

**Review by VlO-OlV — CHANGES_REQUESTED** [review]

File: fictadvisor-api/src/v2/api/controllers/SubjectController.ts [hunk 7]
```diff
@@ -106,48 +79,28 @@ export class SubjectController {
...
+  @ApiEndpoint({
+    summary: 'Delete subject by id',
+    documentation: SubjectDocumentation.DELETE,
+    permissions: PERMISSION.SUBJECTS_DELETE,
   })
   @Delete('/:subjectId')
   delete (
     @Param('subjectId', SubjectByIdPipe) subjectId: string,
-  ) {
+  ): Promise<void> {
```
[the remark this answers is not in the document]
Comment (ReGeNss) [reply fictadvisor-api/src/v2/api/controllers/SubjectController.ts:103]: image
But this endpoint doesn't return anything

**Review by VlO-OlV — APPROVED** [review]

**Pull request merged**